OMCompiler/Compiler/NFFrontEnd/NFType.mo

```diff
@@ -256,6 +256,17 @@ public
     end match;
   end isReal;
 
+  function isRealRecursive
+    input Type ty;
+    output Boolean isReal;
+  algorithm
+    isReal := match ty
+      case REAL() then true;
+      case ARRAY(elementType = REAL()) then true;
```

Comment:
That's no recursion. Did you mean `case ARRAY() then isRealRecursive(ty.elementType);`?

Reply:
yes that is what i intended to do :D

Reply:
A better name would perhaps be `isRealElementType` to conform to the existing names (the function is basically equivalent to `Type.isReal(Type.arrayElementType(ty))`).

Reply:
Yea, but i wanted to be safe if there is any form of recursion e.g. and array containing in an array containing an array etc. And it should also return true for just REAL().

Reply:
Yes, `Type.arrayElementType` returns either the innermost element type of an array type or the type itself.